modin/backends/pandas/query_compiler.py

```diff
@@ -1008,6 +1008,79 @@ def rolling_aggregate(self, rolling_args, func, *args, **kwargs):
         )
         return self.__constructor__(new_modin_frame)
 
+    def unstack(self, level, fill_value):
+        if not isinstance(self.index, pandas.MultiIndex) or (
+            isinstance(self.index, pandas.MultiIndex)
+            and is_list_like(level)
+            and len(level) == self.index.nlevels
+        ):
+            axis = 1
+            new_columns = ["__reduced__"]
+            need_reindex = True
+        else:
+            axis = 0
+            new_columns = None
+            need_reindex = False
+
+        def map_func(df):
+            return pandas.DataFrame(df.unstack(level=level, fill_value=fill_value))
+
+        is_all_multi_list = False
+        if (
+            isinstance(self.index, pandas.MultiIndex)
+            and isinstance(self.columns, pandas.MultiIndex)
+            and is_list_like(level)
+            and len(level) == self.index.nlevels
+            and need_reindex
+        ):
+            is_all_multi_list = True
+            real_cols_bkp = tuple(self.columns)
+            self.columns = [i for i in range(len(self.columns))]
+
+        new_modin_frame = self._modin_frame._apply_full_axis(
+            axis, map_func, new_columns=new_columns
+        )
+        result = self.__constructor__(new_modin_frame)
+
+        if is_all_multi_list:
+            result = result.sort_index()
+            index_ordered = result.index
+            real_cols = list(real_cols_bkp)
+
+            real_cols_extended = []
+            for col in real_cols:
+                real_cols_extended += [col for idx in range(len(self.index))]
```

Comment:
isn't list comprehansion works faster than appending?
```suggestion
            real_cols_extended = [col for col in real_cols for _ in np.arange(len(self.index))]
```

Reply:
Fixed.